src/packages/web-worker/command-handlers/host-call.ts

```diff
@@ -26,47 +29,129 @@ type HostCallResponse =
       error?: unknown;
     };
 
-type ExecuteParams = Parameters<write.Write["execute"]>;
-type RegistersType = ExecuteParams[1];
-type MemoryType = ExecuteParams[2];
+// [KrFr] Gas counter is not used. Can be mocked for now.
+const getGasCounter = (pvm: PvmApiInterface) => {
+  if (isInternalPvm(pvm)) {
+    return pvm.getInterpreter().getGasCounter();
+  }
+
+  return {
+    get: () => {
+      return tryAsGas(pvm.getGasLeft());
+    },
+    set: (gas: Gas) => {
+      if (pvm.setGasLeft) {
+        pvm.setGasLeft(BigInt(gas.toString()));
```

Comment:
```suggestion
        pvm.setGasLeft(BigInt(gas));
```
this should be enough?